src/audio/module_adapter/module/modules.c

```diff
@@ -63,6 +63,7 @@ static int modules_init(struct processing_module *mod)
 	struct comp_dev *dev = mod->dev;
 	const struct ipc4_base_module_cfg *src_cfg = &md->cfg.base_cfg;
 	byte_array_t mod_cfg;
+	bool is_native_sof = false;
```

Comment:
@softwarecki I don't think we have really explained "FDK" in SOF context, so following the git commit messages is a bit challenging. Correct Adrian me if I'm wrong, but in this context "FDK modules" is same as IADK modules which we have explained and documented in https://thesofproject.github.io/latest/architectures/firmware/intel/ace/iadk_modules.html .

So essentially, when you have native SOF loadable modules that don't depend on Intel IADK, the IADK specific code is not needed and code for native modules can be simplified like done in this PR.

I do find the language around "adapter"'s confusing. We have had adapters both between old SOF component.h interface and the new module interface. But we also have adapters between specialized interfaces (like Intel's IADK) and the module interface. This is not incorrect, but makes harder to follow changes like this PR.

Reply:
@kv2019i Yes, i mean IADK modules. FDK modules was rebranded to IADK.

@lyakh The `modules_init()` function is still used, called by Module Adapter. After detecting that it is not an IADK module, the pointer to `module_interface` is overwritten and points to the interface provided by a loaded module. In this way, all other functions of the Processing Module Adapter are omitted.

Reply:
@softwarecki but then - doesn't that mean, that `lib_manager_free_module()` is never called?

Reply:
ok, I think it is a problem, that operations are completely overwritten and `.free()` isn't called any more. But it also looks like that was introduced earlier by https://github.com/thesofproject/sof/commit/a0827526b7e1e4c88060812245a8cb9c381d5745 . I'll block this PR for now, until we clarify or fix this.

Reply:
@lyakh: As you noticed, this PR does not introduce this change, so I see no reason to block it and thus postpone the next PR that will solve the described issue.

Reply:
@softwarecki well, I don't see a fix among currently submitted PRs and I think the offending commit should be reverted as per #8923 instead of being further deepened. Let's revert that commit and then we can base further work on modifying the shim on top of a working state.